crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -33,15 +40,17 @@ pub(super) struct TimemarkedTransaction {
     tx_hash: [u8; 32],
     time_first_seen: Instant,
     address: [u8; 20],
+    cost: HashMap<IbcPrefixed, u128>,
 }
 
 impl TimemarkedTransaction {
-    pub(super) fn new(signed_tx: Arc<SignedTransaction>) -> Self {
+    pub(super) fn new(signed_tx: Arc<SignedTransaction>, cost: HashMap<IbcPrefixed, u128>) -> Self {
```

Comment:
`new` methods should almost always only contain one argument. I think in this case it would be acceptable to leave the method as is but provide a new inherent method `TimemarkedTransaction::set_cost_map(&mut self, cost_map: HashMap<IbcPrefixed, u128>)`

Reply:
I'll add the internal function on the stacked PR [1411](https://github.com/astriaorg/astria/pull/1411) since we do change the cost var there